crates/uv/tests/workspace.rs

```diff
@@ -1440,7 +1439,7 @@ fn workspace_unsatisfiable_member_dependencies_conflicting_dev() -> Result<()> {
     ----- stderr -----
     Using Python 3.12.[X] interpreter at: [PYTHON-3.12]
       × No solution found when resolving dependencies:
-      ╰─▶ Because bar depends on bar and bar depends on anyio==4.2.0, we can conclude that bar depend on anyio==4.2.0.
+      ╰─▶ Because bar depends on bar:dev and bar depends on anyio==4.2.0, we can conclude that bar depend on anyio==4.2.0.
```

Comment:
I cannot figure out why this doesn't say `bar:dev depends on`

Reply:
The tree is wrong

```
      root==0a0.dev0 depends on foo*
        root==0a0.dev0 depends on bar*
          foo==0.1.0 depends on anyio==4.1.0
            bar==0.1.0 depends on anyio==4.2.0
            bar==0.1.0 depends on bar:dev==0.1.0
```

Reply:
And is wrong before any transformation

```
      root==0a0.dev0 depends on foo*
        root==0a0.dev0 depends on bar*
          no versions of foo<0.1.0 | >0.1.0
            foo==0.1.0 depends on anyio==4.1.0
              bar==0.1.0 depends on anyio==4.2.0
                bar==0.1.0 depends on bar:dev==0.1.0
                no versions of bar<0.1.0 | >0.1.0
```

Reply:
This was a display bug https://github.com/astral-sh/uv/pull/6123/files#diff-f8891b530b9722db41a1ef9461f0c2608f639ace189f57ec00e002bf02d7bd00